### ISSUE TITLE
pdf-tools hangs emacs

### ISSUE THREAD

**Issue by sprig** [politza/pdf-tools#252]
Hi!

When opening small files pdf-tools works fine. However, when trying to use on a [medium sized (~30 pages) file](https://github.com/politza/pdf-tools/files/795844/10.1.1.42.8044.pdf), emacs becomes unresponsive for at least several minutes, at which point I kill it. I thought that one of the main benefits of pdf-tools was that it renders pages on the fly, which should presumably allow it to open unlimited-size files. Maybe I have misunderstood however, so please correct me if I'm wrong.

I'm on OSX 10.12

Emacs version: GNU Emacs 25.1.1 (x86_64-apple-darwin16.3.0, NS appkit-1504.76 Version 10.12.2 (Build 16C67)) of 2017-01-14
pdf-tools version: melpa 20170130.

Let me know what other info I can provide.

**Comment by sprig** [comment]
Correction: the problem does not appear to be due to the size of the pdf but rather due to the pdf itself - I edited the attached pdf and removed all but the first two pages. Now it takes a while but the page does render. However, following this emacs becomes unresponsive.

**Comment by politza** [comment]
First try the following and see if the problem persists.

    $ emacs -Q
    M-x package-initialize RET
    M-x pdf-tools-install RET
    C-x C-f /path/to/broken.pdf RET

**Comment by sprig** [comment]
The resolution could be better but it works. Is there a better way to debug this other than binary search?

**Comment by sprig** [comment]
Ok, that was waaay quicker to debug than I anticipated, but disabling global-linum-mode fixed the problem. The reason I even suspected it in the first place is that when I tried opening the small file with regular docview, the last message before emacs hanging was something to do with linum.

It causes problems in some other packages as well (notably org-mode). I guess I'll just disable it altogether. Thanks!
